utils/nctl/scripts/config_gen.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+
+# pylint: disable=missing-module-docstring
+# pylint: disable=line-too-long
+
+import argparse
+import logging
+import pprint as pp
+import sys
+from pathlib import Path
+from tomlkit import dumps
+from tomlkit import loads
+
+
+logger = logging.getLogger(__name__)
+
+
+def main():
+    """The meat and potatoes."""
+    args = parse_args()
+    logger = logger_setup(args.log_level)
+    logger.info("Input args: %s", vars(args))
+
+    logger.info("Opening overrides toml file: %s", args.override_file)
+    override_dict = load_toml_file(args.override_file)
+
+    logger.info("Opening toml file: %s", args.toml_file)
+    toml_dict = load_toml_file(args.toml_file)
+
+    logger.info("Applying overrides to toml file")
+    new_toml_dict = edit_toml(override_dict, toml_dict)
+
+    if args.output_file is not None:
+        logger.info("Generating toml file to: %s", args.output_file)
+        write_toml(args.output_file, new_toml_dict)
+    else:
+        pp.pprint(new_toml_dict)
+
+    logger.info("Finished!")
+
+
+def logger_setup(level):
+    """Sets up logger based on based in log level."""
+    level = level.lower()
+
+    if level == "debug":
+        log_format = (
+            "[%(asctime)s] %(levelname)s [%(name)s.%(funcName)s:%(lineno)d] %(message)s"
+        )
+    else:
+        log_format = "[%(asctime)s] %(levelname)s %(message)s"
+
+    levels = {"info": logging.INFO, "debug": logging.DEBUG, "warning": logging.WARNING}
+    level = levels.get(level)
+
+    logging.basicConfig(
+        stream=sys.stdout, level=level, format=log_format, datefmt="%d-%b-%Y %H:%M:%S"
```

Comment:
(I'm a fan of [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601), so I'd say `%Y-%m-%d`. But non-believers will argue it's just a matter of taste.)

Reply:
> Looks great! 👍
> 
> I didn't go through all the chainspecs and overrides in detail in the review: Did you create them programmatically, or compare that the generated files with the overrides applied are identical to the files that were removed?

I diffed each of them against what was in `resources/local/*.toml`  and pulled out what was being changed.

Reply:
I agree with ya: https://github.com/casper-network/casper-node/pull/3159/commits/87433a5a65b334c9d84cd322f7c9c421db5b5e98